### PR TITLE
Make changes in the extractor for the new workflow

### LINKED ISSUE #153: New trigger comment for Workflow to reuse changed Inputs of Pull Request
https://docs.google.com/document/d/1_tuJsvtgA5ejtDajOO7ndCjrYzxZMllCPsjMkAum7Oc/edit

One of most desirable feature is to get config based on what is changed in code of Module/Check.
As we demand full coverage, all configs that matters will be at Input files of diff on change

Example ("git show --stat HEAD”): 
```
https://github.com/checkstyle/checkstyle/commit/73120c4943ce07533e812a6b241869fca7eade80 
https://github.com/checkstyle/checkstyle/commit/73120c4943ce07533e812a6b241869fca7eade80.patch 
 .../whitespace/MethodParamPadCheck.java       |  3 ++
 .../checks/whitespace/MethodParamPadCheck.xml |  2 +-
 src/main/resources/google_checks.xml          |  2 +-
 .../whitespace/MethodParamPadCheckTest.java   | 17 +++++++-
 .../InputMethodParamPadRecords.java           |  2 +-
 .../InputMethodParamPadCheckConstructors.java | 40 +++++++++++++++++++
 .../InputMethodParamPadWhitespaceAround.java  |  3 +-
 .../checks/whitespace/methodparampad.xml      |  4 ++
```

So our target is:
```
 .../InputMethodParamPadRecords.java
 .../InputMethodParamPadCheckConstructors.java
 .../InputMethodParamPadWhitespaceAround.java
```

We extract configs from it and can use for execution, list of projects will be some default. 

Usage for user: 

Github, generate report for configs in PR description
It will simply use configs as it was defined in original workflow, this will be in use for complicated cases when we need different configs for master and patch.

`Github, generate report for config from <path to Input in checkstyle repo>`
`Github, generate report by config from src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPadWhitespaceAround.java`
`Github, generate report by config from InputMethodParamPadWhitespaceAround.java`

Workflow just clone PR. Workflow takes config out of such Input and generate config.xml.
Workflow use some default list of projects file.
And simply executes diff report based on configs files that are just generated in execution. Diff.groovy already has path to checkstyle repo of branch, it just need to generate full xml from Input same/similar way as it is done in “extractor”. “Diff.groovy” should have special argument to define this mode of execution.
No need to publish generated content, it will be part of report. 
look at report: https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/e6cdb69_2024152818/reports/diff/index.html
it has all configs that user provided, first two links in index.html
If there is duplicate file name (same name in different folders), workflow will fail.

In code of workflow we can check that config file is ending with “*.java” and run extractor over it to get ready to use xml for diff.groovy. So it is just extra call for extractor before regular  diff.groovy call.
 
File search will be linux util based (“find . -path”):
```
✔ ~/java/github/romani/checkstyle [master|⚑ 2] 
$ find . -path "*InputMethodParamPadWhitespaceAround.java"
./src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPadWhitespaceAround.java

✔ ~/java/github/romani/checkstyle [master|⚑ 2] 
$ find . -path "*methodparampad/InputMethodParamPadWhitespaceAround.java"
./src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPadWhitespaceAround.java
```

`Github, generate report for config from <path to Input in checkstyle repo> , list of targets <http link to raw content>`
`Github, generate report by config from src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPadWhitespaceAround.java , list of targets https://raw.githubusercontent.com/checkstyle/test-configs/main/SummaryJavadoc/Example1/list-of-projects.properties`

### PR DESCRIPTION
**Pull request by relentless-pursuit**
part of #153

### CODE REVIEW

**Comment by romani** [comment]
Please put in commit message exact reason of update, kind of summary of update

**Comment by relentless-pursuit** [comment]
> Please put in commit message exact reason of update, kind of summary of update

Updated

**Comment by relentless-pursuit** [comment]
@romani can we merge this. and release the extractor. i can then start testing the workflow on 5 scenarios i created before merging.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: extractor/src/main/java/com/example/extractor/CheckstyleExampleExtractor.java [hunk 1]
```diff
@@ -137,13 +155,36 @@ public static void main(final String[] args) throws Exception {
...
+
+            final byte[] buffer = new byte[BUFFER_SIZE];
+            int length = inputStream.read(buffer);
+            while (length > 0) {
+                outputStream.write(buffer, 0, length);
+                length = inputStream.read(buffer);
+            }
+        }
+        catch (IOException ex) {
+            LOGGER.log(Level.SEVERE, "Error outputting default projects list", ex);
```
[the remark this answers is not in the document]
Comment (relentless-pursuit) [reply extractor/src/main/java/com/example/extractor/CheckstyleExampleExtractor.java]: done

**Pull request merged**